packages/arb-token-bridge-ui/src/hooks/useClaimWithdrawal.ts

```diff
@@ -19,20 +19,22 @@ import { fetchErc20Data } from '../util/TokenUtils'
 import { fetchNativeCurrency } from './useNativeCurrency'
 
 export type UseClaimWithdrawalResult = {
-  claim: (tx: MergedTransaction) => Promise<void>
+  claim: () => Promise<void>
   isClaiming: boolean
 }
 
-export function useClaimWithdrawal(): UseClaimWithdrawalResult {
+export function useClaimWithdrawal(
+  tx: MergedTransaction
+): UseClaimWithdrawalResult {
   const {
     app: { arbTokenBridge }
   } = useAppState()
   const { address } = useAccount()
-  const { data: signer } = useSigner()
+  const { data: signer } = useSigner({ chainId: tx.parentChainId })
```

Comment:
Now users don't have to do 2 clicks to claim + redeem. It happens in the same flow, and that's why we have to pass the correct chain ID to the signer or otherwise it will be a signer for the old network (before the switch happened)